entrypoint.sh

```diff
@@ -15,6 +15,61 @@ postconf -F '*/*/chroot = n'
 postconf -x mydestination
 [ "$DEBUG" == "true" ] && sed -i 's/smtpd$/smtpd -v/g' /etc/postfix/master.cf
 
+if [ ${DKIM_ENABLED:-false} == "true" ]; then
+    if [ ! -f /opt/__dkim_init ]; then
+        [ -z "${DKIM_SELECTOR:-}" ] && DKIM_SELECTOR='default'
+        if [ ! -f "/etc/opendkim/keys/${DKIM_DOMAIN}/${DKIM_SELECTOR}.private" ]; then
+            echo "Error! No such private DKIM KEY found /etc/opendkim/keys/${DKIM_DOMAIN}/${DKIM_SELECTOR}.private ! Aborting..."
+            exit 1
+        fi
+        cat >>/etc/opendkim.conf <<EOL
+Canonicalization   relaxed/simple
+Mode               sv
+SubDomains         no
+AutoRestart         yes
+AutoRestartRate     10/1M
+Background          yes
+DNSTimeout          5
+SignatureAlgorithm  rsa-sha256
+Socket inet:8891
+#OpenDKIM user
+# Remember to add user postfix to group opendkim
+UserID             opendkim
+# Map domains in From addresses to keys used to sign messages
+KeyTable           refile:/etc/opendkim/KeyTable
+SigningTable       refile:/etc/opendkim/SigningTable
+# Hosts to ignore when verifying signatures
+ExternalIgnoreList  /etc/opendkim/TrustedHosts
+# A set of internal hosts whose mail should be signed
+InternalHosts       /etc/opendkim/TrustedHosts
+LogWhy                  yes
+Syslog                  yes
+SyslogSuccess           yes
+EOL
+
+        echo "Domain ${DKIM_DOMAIN}" >>/etc/opendkim.conf
+        echo 'RequireSafeKeys False' >>/etc/opendkim.conf
+        mkdir -p /etc/opendkim
+        echo "*@${DKIM_DOMAIN} ${DKIM_SELECTOR}._domainkey.${DKIM_DOMAIN}" >>/etc/opendkim/SigningTable
+        echo "${DKIM_SELECTOR}._domainkey.${DKIM_DOMAIN} ${DKIM_DOMAIN}:${DKIM_SELECTOR}:/etc/opendkim/keys/${DKIM_DOMAIN}/${DKIM_SELECTOR}.private" >>/etc/opendkim/KeyTable
+        [ -z "${DKIM_AUTHORIZED_HOSTS:-}" || echo ${DKIM_AUTHORIZED_HOSTS} | sed 's/,/\n/g' | xargs -L 1 >>/etc/opendkim/TrustedHosts
+        echo "*.${DKIM_DOMAIN}" >>/etc/opendkim/TrustedHosts
+        chown opendkim:opendkim /etc/opendkim/keys -R
+        [ -e /var/spool/postfix/opendkim ] || mkdir /var/spool/postfix/opendkim
+        chown opendkim:postfix /var/spool/postfix/opendkim
+        sed -i 's|local:/run/opendkim/opendkim.sock|local:/var/spool/postfix/opendkim/opendkim.sock|g' /etc/opendkim.conf
+        cat >>/etc/postfix/main.cf <<EOL
+# Milter configuration
+milter_default_action = accept
+milter_protocol = 2
+smtpd_milters = inet:localhost:8891
```

Comment:
is it something that may change?

Reply:
no it is an internal socket server of Opendkim to answer postfix about signature requests